h2o-algos/src/main/java/hex/tree/SharedTreeModel.java

```diff
@@ -736,5 +738,110 @@ public void generate(JCodeSB out) {
   protected <T extends JCodeSB> T toJavaForestName(T sb, String mname, int t ) {
     return (T) sb.p(mname).p("_Forest_").p(t);
   }
+  
+  @Override
+  public Frame scoreContributions(Frame frame, Key<Frame> destination_key) {
+    if (_output.nclasses() > 2) {
+      throw new UnsupportedOperationException(
+              "Calculating contributions is currently not supported for multinomial models.");
+    }
+    if ((!_parms.algoName().equals("GBM")) && (!_parms.algoName().equals("DRF"))) {
```

Comment:
there are certainly more elegant and less expensive ways of achieving the same result. 
why I dont like it:
1) string comparison to determine functionality
2) IsoFor implements a method that throws UnsupportedOp - this will surprise someone
few ideas how to improve:
1) create a SharedTreeModel subclass that also implements Contributions and contains this method, have GBM and DRF inherit from that
2) move this method to a separate class that only implements Contributions, use this new class in GBM and DRF model classes scoreContributions method.

Reply:
I have addressed this here (hopefully): 2e5b7fa